crates/miden-objects/src/block/account_tree.rs

```diff
@@ -41,68 +260,9 @@ impl AccountTree {
 
-    /// Creates a new, empty account tree.
-    pub fn new() -> Self {
-        AccountTree { smt: Smt::new() }
-    }
-
-    /// Returns a new [`Smt`] instantiated with the provided entries.
-    ///
-    /// If the `concurrent` feature of `miden-crypto` is enabled, this function uses a parallel
-    /// implementation to process the entries efficiently, otherwise it defaults to the
-    /// sequential implementation.
-    ///
-    /// # Errors
-    ///
-    /// Returns an error if:
-    /// - the provided entries contain multiple commitments for the same account ID.
-    /// - multiple account IDs share the same prefix.
-    pub fn with_entries<I>(
-        entries: impl IntoIterator<Item = (AccountId, Word), IntoIter = I>,
-    ) -> Result<Self, AccountTreeError>
-    where
-        I: ExactSizeIterator<Item = (AccountId, Word)>,
-    {
-        let entries = entries.into_iter();
-        let num_accounts = entries.len();
-
-        let smt = Smt::with_entries(
-            entries.map(|(id, commitment)| (Self::id_to_smt_key(id), commitment)),
-        )
-        .map_err(|err| {
-            let MerkleError::DuplicateValuesForIndex(leaf_idx) = err else {
-                unreachable!("the only error returned by Smt::with_entries is of this type");
-            };
-
-            // SAFETY: Since we only inserted account IDs into the SMT, it is guaranteed that
-            // the leaf_idx is a valid Felt as well as a valid account ID prefix.
-            AccountTreeError::DuplicateStateCommitments {
-                prefix: AccountIdPrefix::new_unchecked(
-                    Felt::try_from(leaf_idx).expect("leaf index should be a valid felt"),
-                ),
-            }
-        })?;
-
-        // If the number of leaves in the SMT is smaller than the number of accounts that were
-        // passed in, it means that at least one account ID pair ended up in the same leaf. If this
-        // is the case, we iterate the SMT entries to find the duplicated account ID prefix.
-        if smt.num_leaves() < num_accounts {
-            for (leaf_idx, leaf) in smt.leaves() {
-                if leaf.num_entries() >= 2 {
-                    // SAFETY: Since we only inserted account IDs into the SMT, it is guaranteed
-                    // that the leaf_idx is a valid Felt as well as a valid
-                    // account ID prefix.
-                    return Err(AccountTreeError::DuplicateIdPrefix {
-                        duplicate_prefix: AccountIdPrefix::new_unchecked(
-                            Felt::try_from(leaf_idx.value())
-                                .expect("leaf index should be a valid felt"),
-                        ),
-                    });
-                }
-            }
-        }
-
-        Ok(AccountTree { smt })
+    /// Creates a new `AccountTree` from its inner representation.
+    pub fn new(smt: S) -> Self {
+        AccountTree { smt }
```

Comment:
We don't validate anything here because we assume that implementors of `AccountTreeBackend` know what they're doing?

If so, I'd still mention something like this in the docs:
```rust
/// # Warning
///
/// Assumes the provided SMT upholds the guarantees of the [`AccounTree`]. See type-level docs for details.
```

Any I would probably even rename this to `new_unchecked` then.